src/components/hack-details.js

```diff
@@ -15,52 +16,33 @@ import {
 import { makeStyles } from "@material-ui/core/styles";
 import ErrorOutlineOutlinedIcon from "@material-ui/icons/ErrorOutlineOutlined";
 
-import { connect } from "react-redux";
-
 const useStyles = makeStyles(theme => ({
   root: {
     margin: theme.spacing(2),
-    padding: theme.spacing(2)
+    padding: theme.spacing(2),
+    width: "80vw"
   },
   rightFeild: {
     textAlign: "right"
   },
   loading: {
     marginTop: theme.spacing(10)
   },
-  chipMArgin: {
-    marginRight: theme.spacing(1)
+  marginTop: {
+    marginTop: theme.spacing(2)
   }
 }));
 
-export function HackDetails(props) {
+export function HackDetails({ match, dispatch, userId, hackDetails }) {
   const classes = useStyles();
 
-  const [data, setData] = useState(null);
-
   useEffect(() => {
-    //console.log(props.match.params);
-
-    if (props.match !== undefined) {
-      Axios.get(
-        UrlJoin(process.env.REACT_APP_API_URL,"hackdetail/",props.match.params.id)
-      ).then(res => {
-        setData(res.data)
-      });
+    if (!hackDetails || hackDetails._id !== match.params.id) {
+      dispatch(fetchingHackDetails(match.params.id));
     }
-  }, [props.match]);
-
-  const joinHack = event => {
-    const object = {
-      hackId: props.match.params.id,
-      userId: props.userId
-    };
-    Axios.post(UrlJoin(process.env.REACT_APP_API_URL,"joinhack"), object).then(res => {
-      setData(res.data);
-    });
-  };
+  }, [hackDetails, dispatch, match]);
 
-  if (data !== null) {
+  if (hackDetails) {
```

Comment:
if you've edited one hack then edit another, will it briefly show the details of the first hack while its loading the second?